internal/controller/bmc/pod.go

```diff
@@ -0,0 +1,128 @@
+/*
+ * This file is part of the KubeVirt/KubeVirtBMC project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+package bmc
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	bmcv1beta1 "kubevirt.io/kubevirtbmc/api/bmc/v1beta1"
+)
+
+type AgentImageConfig struct {
+	ImageName     string
+	ImageTag      string
+	ContainerName string
+	FullImage     string
+}
+
+func NewAgentImageConfig() AgentImageConfig {
+	registryHost := os.Getenv("REGISTRY_HOST")
+	if registryHost == "" {
+		registryHost = "registry"
+	}
+
+	registryPort := os.Getenv("REGISTRY_PORT")
+	if registryPort == "" {
+		registryPort = "5000"
+	}
+
+	imageRepo := os.Getenv("VIRTBMC_IMAGE_NAME")
+	if imageRepo == "" {
+		imageRepo = "virtbmc"
+	}
+
+	imageTag := os.Getenv("VIRTBMC_IMAGE_TAG")
+	if imageTag == "" {
+		imageTag = "latest"
+	}
+	registry := fmt.Sprintf("%s:%s", registryHost, registryPort) // e.g. "registry:5000"
+	imageName := fmt.Sprintf("%s/%s", registry, imageRepo)       // e.g. "registry:5000/virtbmc"
+
+	return AgentImageConfig{
+		ImageName:     imageName,
+		ImageTag:      imageTag,
+		ContainerName: imageRepo,
+		FullImage:     fmt.Sprintf("%s:%s", imageName, imageTag), // e.g. "registry:5000/virtbmc:latest"
+	}
+}
+
+func (r *VirtualMachineBMCReconciler) NewPodSpec(bmc *bmcv1beta1.VirtualMachineBMC) corev1.PodSpec {
+	return corev1.PodSpec{
+		Containers: []corev1.Container{
+			{
+				Name:            r.AgentImageName.ContainerName,
+				Image:           r.AgentImageName.FullImage,
+				ImagePullPolicy: corev1.PullAlways,
+				Args: []string{
+					"--address",
+					"0.0.0.0",
+					"--ipmi-port",
+					strconv.Itoa(IpmiContainerPort),
+					"--redfish-port",
+					strconv.Itoa(RedfishContainerPort),
```

Comment:
We could make this configurable, and also we could add configurables to the API that allow enabling/disabling IPMI and Redfish individually. But in a future PR.

Reply:
This is a good enhancement!

@salawhaaat Do you want to create an issue to track it? Or I can do it.